lib/handler.js

```diff
@@ -68,6 +70,13 @@ const handler = async (req, res, pgPool) => {
       res,
       pgPool,
       fetchMinersRSRSummary)
+  } else if (req.method === 'GET' && segs[0] === 'nodes' && segs[1] === 'daily' && segs.length === 2) {
+    await getStatsWithFilterAndCaching(
+      pathname,
+      searchParams,
+      res,
+      pgPool,
+      fetchDailyNodeMetrics)
```

Comment:
We are very aggressively caching the responses from the stats API. Changing the response format requires us to nuke the Cloudflare CDN cache manually. Let's try to get this endpoint right on the first try.

When I look at the SQL query, we return a list of all station IDs seen on a given day.  @PatrickNercessian: How do you envision visualising this data?

I propose to approach the spark-stats API changes from the perspective of the dashboard we are going to build and tailor the data to the needs of those charts.

For example, I can see how it's useful to show how the network size (number of unique Station IDs) evolves over time. I have already implemented such charts for participants (unique wallet addresses). Can we adopt that code for station instances too? (I think it's ok to simply copy&paste. There is also an option to extract shared helpers parameterised by the table name, but that could end up being too complex and not worth the effort right now.) See

- `GET /participants/daily`: 
  https://github.com/filecoin-station/spark-stats/blob/906d6e2788f4ef74389a111c2a6dbdcbbf9106c6/lib/handler.js#L50
- `GET /participants/monthly`: 
  https://github.com/filecoin-station/spark-stats/blob/906d6e2788f4ef74389a111c2a6dbdcbbf9106c6/lib/handler.js#L57
- upcoming `GET /participants/churn-rate`: 
  https://github.com/filecoin-station/spark-stats/pull/74

Reply:
> there should be two new files - one for the production code and one for the tests - that can be copied as-is from spark-stats to voyager-stats to mirror the changes.

Makes sense! I did this for platform-stats-fetcher.js already, do you think we should do it for handler code as well? I wasn't sure how we could accomplish that when I was implementing.

> For example, I can see how it's useful to show how the network size (number of unique Station IDs) evolves over time.

I also visualized the data this way, as you are describing it here.

My thought process was to have one endpoint that serves the data needed by Grafana, and Grafana can handle the logic with how it wants to display that data (using Grafana Transformations, which we already plan to use to support multiple Infinity data sources and merge them). This way we don't need to tailor our API to any specific visualizations, and we can just serve the data to Grafana. And we also avoid these cache issues you're describing, since it would instead just be a modification of the Grafana Transformation instead of the underlying API endpoint and its cache. What do you think?

Reply:
As we discussed in our call, I am concerned about the egress costs of this solution, but I am happy to try it!

Considering that the new route uses a different path now, allowing us to create `/stations/daily` in the future, if it turns out we cannot use your proposed solution, I am happy to test this in production by landing & deploying the pull request.